### ISSUE TITLE
Initial D Arcade Stage (Export, Rev A) [initdexp.zip]: soft-crash before starting the race + "Insert Card" not working

### ISSUE THREAD

**Issue by cinnamonmatexfce** [flyinghead/flycast#1337]
Platform / OS / Hardware:
 - *$ uname -rmo*
   - ```
     6.6.4-arch1-1 x86_64 GNU/Linux
     ```
 - *$ lspci -nn | grep VGA*
   - ```
      0a:00.0 VGA compatible controller [0300]: NVIDIA Corporation TU116 [GeForce GTX 1660 SUPER] [10de:21c4] (rev a1)
     ```

Github hash:
 - *MASTER*
	`$ cd flycast-master && git rev-parse HEAD`
	5fde889d7ca30cc0059eaf64816878df2cb9f898

 - *DEV*
	`$ cd flycast-dev && git rev-parse HEAD`
	05cd7ee8d383204085b6362e268ef60e3acf1f54

&nbsp;

 ### SOFT-CRASH
 #### `dev` and `master`
 \[\[ [flycast-x86_64.AppImage, v2.2](https://github.com/flyinghead/flycast/releases/tag/v2.2) does NOT *soft-crash* \]\]
Flycast (both `dev` and `master`) *soft-crashes* itself just before loading race stage.
Screenshot below here reports the last working step:
![initdexp zip--before-soft-crash](https://github.com/flyinghead/flycast/assets/65762930/3a4046ca-552b-4a3c-882e-2f74204b4f62).

 ### INSERT CARD
 #### `v2.2` and `dev` and `master`
Also, *Insert Card* seems it's not working (I must purchase a new card **every time** I start the game); this happens in `v2.2` and `dev` and `master`.
That's odd, added to the fact that the *KICKBAK INITIALIZE* procedure, happens correctly when launching the game for the **first time**:
![initdexp zip--card-kickback](https://github.com/flyinghead/flycast/assets/65762930/d9d801d6-21fe-41b4-889f-607170cee183)

___

\~\~\~ I don't think that's a bug, I'm reporting it just for letting you know \~\~\~
 - this warning appears when launching the game for the **first time** only (NOT in `v2.2`, though):
   `00:12:923 hw/naomi/naomi_flashrom.cpp:136 W[NAOMI]: EEPROM record doesn't exist or is too short`
```
$ flycast initd/initdexp.zip 
00:00:229 sdl/sdl.cpp:629 N[RENDERER]: Monitor refresh rate: 60 Hz (1920 x 1080)
00:00:270 rend/gui.cpp:301 N[RENDERER]: Screen DPI is 95, size 640 x 480. Scaling by 1.00
00:00:806 rend/gles/gles.cpp:577 N[RENDERER]: OpenGL version 4.3
00:00:814 hw/mem/addrspace.cpp:431 N[VMEM]: Info: nvmem is enabled
00:00:860 hw/mem/addrspace.cpp:462 N[VMEM]: BASE 0x7f4fac940000 RAM(16 MB) 0x7f4fb8940000 VRAM64(8 MB) 0x7f4fb0940000 ARAM(2 MB) 0x7f4fcc940000
00:00:882 hw/mem/addrspace.cpp:431 N[VMEM]: Info: nvmem is enabled
00:00:924 hw/mem/addrspace.cpp:462 N[VMEM]: BASE 0x7f4fac940000 RAM(32 MB) 0x7f4fb8940000 VRAM64(16 MB) 0x7f4fb0940000 ARAM(8 MB) 0x7f4fcc940000
00:12:897 hw/naomi/naomi_cart.cpp:643 N[NAOMI]: NAOMI GAME ID [INITIAL D] region fe players 1 vertical 1
00:12:901 emulator.cpp:54 N[BOOT]: Game ID is [INITIAL D]
00:12:923 hw/naomi/naomi_flashrom.cpp:148 N[NAOMI]: Initializing Naomi EEPROM for game INITIAL D                       
00:12:923 hw/naomi/naomi_flashrom.cpp:207 N[NAOMI]: EEPROM: horizontal monitor orientation
00:12:923 emulator.cpp:54 N[BOOT]: Game ID is [INITIAL D]
00:12:923 hw/naomi/naomi_flashrom.cpp:136 W[NAOMI]: EEPROM record doesn't exist or is too short
$ flycast initd/initdexp.zip 
00:00:159 sdl/sdl.cpp:629 N[RENDERER]: Monitor refresh rate: 60 Hz (1920 x 1080)
00:00:178 rend/gui.cpp:301 N[RENDERER]: Screen DPI is 95, size 640 x 480. Scaling by 1.00
00:00:355 rend/gles/gles.cpp:577 N[RENDERER]: OpenGL version 4.3
00:00:357 hw/mem/addrspace.cpp:431 N[VMEM]: Info: nvmem is enabled
00:00:384 hw/mem/addrspace.cpp:462 N[VMEM]: BASE 0x7f72a0bf0000 RAM(16 MB) 0x7f72acbf0000 VRAM64(8 MB) 0x7f72a4bf0000 ARAM(2 MB) 0x7f72c0bf0000
00:00:392 hw/mem/addrspace.cpp:431 N[VMEM]: Info: nvmem is enabled
00:00:417 hw/mem/addrspace.cpp:462 N[VMEM]: BASE 0x7f72a0bf0000 RAM(32 MB) 0x7f72acbf0000 VRAM64(16 MB) 0x7f72a4bf0000 ARAM(8 MB) 0x7f72c0bf0000
00:07:199 hw/naomi/naomi_cart.cpp:643 N[NAOMI]: NAOMI GAME ID [INITIAL D] region fe players 1 vertical 1
00:07:199 emulator.cpp:54 N[BOOT]: Game ID is [INITIAL D]
00:07:207 hw/naomi/naomi_flashrom.cpp:207 N[NAOMI]: EEPROM: horizontal monitor orientation
00:07:207 emulator.cpp:54 N[BOOT]: Game ID is [INITIAL D]
```

**Comment by flyinghead** [comment]
I can reproduce the freeze with the dev branch but not with master. This is likely due to the changes I pushed to dev yesterday.
Insert Card works fine with both master and dev here. Make sure to complete the game until after the "Game Over" screen appears, since the card is only written at this point. (I added a visible message when the card is ejected on the dev branch.)

**Comment by cinnamonmatexfce** [comment]
>  after the "Game Over" screen appears, since the card is only written at this point. (I added a visible message when the card is ejected on the dev branch.)

Mmm... I was suspecting some sort of missing writes in the card, indeed.
As soon as I can, I'm going to try it and I'll let you know.
Oh, and - as always - thank you for helping us and for your awesome work with *Flycast*.

💪
